File: javascript/SupplementMarketInterface.js
function SupplementMarketInterface()
{};

/**
 * Get tax of product in percentage.
 *
 * @return integer
 */
SupplementMarketInterface.prototype.getTax = function(product)
{
    throw Error('Implement the method getTax');
};

/**
 * Get offer of product if exists.
 *
 * If the product has offer, it will return a array where will indicate what offer apply. Ej. 2x1 -> array(2, 1), 3x2 -> array(3,2).
 * If the product does not have offer, it will return null.
 *
 * @return array|null
 */
SupplementMarketInterface.prototype.getOffer = function(product)
{
    throw Error('Implement the method getOffer');
};

File: java/scm.kata.supermarket/src/SupplementMarketInterface.java
import java.util.List;

/**
 * Interface to report what supplement have the products in Market.
 */
public interface SupplementMarketInterface {
    /**
     * Get tax of product in percentage.
     *
     * @param product
     * @return integer
     */
    int getTax(String product);

    /**
     * Get offer of product if exists.
     * <p/>
     * If the product has offer, it will return a array where will indicate what offer apply. Ej. 2x1 -> List(2, 1), 3x2 -> List(3,2).
     * If the product does not have offer, it will return null.
     *
     * @param product
     * @return List
     */
    List getOffer(String product);
}
